src/components/Payment.js

```diff
@@ -63,6 +77,12 @@ class Payment extends Component {
             )
           ),
           this.renderAmountField(false)('Tip in Credit')
+        ),
+        CloseOrderBtn(
+          {
+            shouldBeDisabled: this.state.balance !== 0,
+            handleClick: this.props.closeOrder.bind.apply(this.props.closeOrder, [null].concat(this.allInputAmounts()))
```

Comment:
if it were to be an amount object, i think you would be able to just pass the object directly here without have to bind anything